packages/wallet-sdk/src/relay/MobileRelay.ts

```diff
@@ -1,41 +1,73 @@
-import { isInIFrame } from "../util";
+import { Observable } from "rxjs";
+
+import { getLocation } from "../util";
+import { RelayMessage, RelayMessageType } from "./RelayMessage";
 import { WalletLinkRelay, WalletLinkRelayOptions } from "./WalletLinkRelay";
 import { CancelablePromise } from "./WalletSDKRelayAbstract";
+import { Web3Method } from "./Web3Method";
+import { Web3Request } from "./Web3Request";
+import { Web3RequestMessage } from "./Web3RequestMessage";
 import { RequestEthereumAccountsResponse } from "./Web3Response";
 
 export class MobileRelay extends WalletLinkRelay {
-  private _useMobileWalletLink: boolean;
+  private _enableMobileWalletLink: boolean;
 
   constructor(options: Readonly<WalletLinkRelayOptions>) {
     super(options);
-    this._useMobileWalletLink = options.useMobileWalletLink ?? false;
+    this._enableMobileWalletLink = options.enableMobileWalletLink ?? false;
   }
 
   // override
   public requestEthereumAccounts(): CancelablePromise<RequestEthereumAccountsResponse> {
-    if (this._useMobileWalletLink) {
+    if (this._enableMobileWalletLink) {
       return super.requestEthereumAccounts();
     }
 
     // TODO: Implement & present in-page wallet picker instead of navigating to www.coinbase.com/connect-dapp
     return {
       promise: new Promise(() => {
-        let location: Location;
-        try {
-          if (isInIFrame() && window.top) {
-            location = window.top.location;
-          } else {
-            location = window.location;
-          }
-        } catch (e) {
-          location = window.location;
-        }
-
+        const location = getLocation();
         location.href = `https://www.coinbase.com/connect-dapp?uri=${encodeURIComponent(
           location.href,
         )}`;
       }),
       cancel: () => {},
     };
   }
+
+  // override
+  protected publishEvent(
+    event: string,
+    message: RelayMessage,
+    callWebhook: boolean,
+  ): Observable<string> {
+    const observable = super.publishEvent(event, message, callWebhook);
+
+    if (
+      this._enableMobileWalletLink &&
+      event === "Web3Request" &&
+      message.type === RelayMessageType.WEB3_REQUEST
+    ) {
+      const web3RequestMessage = message as Web3RequestMessage;
+      this.openCoinbaseWalletDeeplink(web3RequestMessage.request);
```

Comment:
note that there might be a potential edge case in this code. 
we have overridden the publishEvent function to open the cbwallet app whenever a client sends web3 requests to walletlink. (filtering out SESSION_ID_REQUEST and LINKED messages) ideally, this logic should appended to the super.observable.
however, we have encountered an issue where the browser seems to restrict calling window.open within the rxjs subscription block. so it's called when the observable is created.

```
      return observable.pipe(
        tap(() => {
          console.log("MobileRelay publishEvent tap within tap");
          this.openCoinbaseWalletDeeplink(web3RequestMessage.request);
        }),
      );

```